hooks/post-command

```diff
@@ -1,62 +1,44 @@
 #!/bin/bash
-# shellcheck disable=SC2001
-set -euo pipefail
+set -ue
 
-if [[ "${BUILDKITE_PLUGIN_CACHE_DEBUG:-false}" =~ (true|on|1) ]] ; then
-  set -x
+DIR="$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd)"
+
+# shellcheck disable=SC1090
+. "$DIR/../lib/shared.bash"
+
+# get bucket name from org slug ie.. chef-oss or chef
+if [[ "${BUILDKITE_ORGANIZATION_SLUG}" == "chef-oss" ]]; then
+  bucket_name="core-buildkite-cache-chef-oss-prod"
+elif [[ "${BUILDKITE_ORGANIZATION_SLUG}" == "chef" ]]; then
+  bucket_name="core-buildkite-cache-chef-prod"
+else
+  bucket_name="core-buildkite-cache-chef-canary-canary"
 fi
 
-if [[ -n "${BUILDKITE_PLUGIN_CACHE_CACHE_KEY:-}" ]] ; then
+paths=( $(plugin_read_list CACHED_FOLDERS) )
 
-  cache_key_prefix=$(echo "$BUILDKITE_PLUGIN_CACHE_CACHE_KEY" | sed -e 's/{.*//')
-  template_value=$(echo "$BUILDKITE_PLUGIN_CACHE_CACHE_KEY" | sed -e 's/^[^\{{]*[^A-Za-z]*//' -e 's/.}}.*$//' | tr -d \' | tr -d \")
+for path in "${paths[@]}"
+do
+  # if the path is bad - skip it, don't break things
+  if [[ -d ${path} ]]; then
+    filename=`echo ${path} |sed 's/\//_/g'`
 
-  if [[ $template_value == *"checksum"* ]]; then
-    checksum_argument=$(echo "$template_value" | sed -e 's/checksum*//')
-    function=${template_value/"checksum"/"shasum"}
-    result=$($function | tr -d "$checksum_argument")
-    cache_key="$cache_key_prefix$result"
-  else
-    cache_key=$BUILDKITE_PLUGIN_CACHE_CACHE_KEY
-  fi
+    # remove trailing '-' from filename
+    if [[ ${filename: -1} == "_" ]]; then
+      filename=${filename::${#filename}-1}
+    fi
 
-  paths=()
+    echo "--- compressing ${path} into tar"
+    tar -czf ${filename}.tar.gz ${path}
 
-  if [[ -n "${BUILDKITE_PLUGIN_CACHE_PATHS:-}" ]] ; then
-    paths+=("$BUILDKITE_PLUGIN_CACHE_PATHS")
-  fi
+    # remove '/' from label if it exists - only character that breaks S3 pathing
+    BUILDKITE_LABEL=`echo ${BUILDKITE_LABEL} |sed 's/\///g'`
```

Comment:
we have to be careful with `sed` as it can sometimes behave differently on Mac vs Linux. It _looks_ okay, but we'll want to watch out.

Reply:
oh interesting, isn't bash the same throughout? or do different OS have different versions?

Reply:
so `sed` is actually a linux utility, not a bash function. Since it is a utility, it can differ from OS to OS. Darwin uses BSD sed, not GNU sed.